server/mediaserver.go

```diff
@@ -484,22 +503,25 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 
 func endRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.RTMPVideoStream) error {
 	return func(url *url.URL, rtmpStrm stream.RTMPVideoStream) error {
-		rtmpID := rtmpStrm.GetStreamID()
+		mid, err := rtmpManifestID(rtmpStrm)
+		if err != nil {
+			glog.Error("Ending stream with invalid manifest ID ", err)
+			return err
+		}
 		//Remove RTMP stream
-		delete(s.rtmpStreams, core.StreamID(rtmpID))
-		if s.CurrentPlaylist != nil {
-			s.CurrentPlaylist.Cleanup()
-			s.CurrentPlaylist = nil
+		s.connectionLock.Lock()
+		defer s.connectionLock.Unlock()
+		cxn, ok := s.rtmpConnections[mid]
+		if !ok || cxn.pl == nil {
```

Comment:
Are there cases where we could have a `rtmpConnection` without a `core.PlaylistManager`? Or is this just for extra safety?

Reply:
This check is just for safety -- at the moment, there shouldn't be any cases where the PlaylistManager is empty.